src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+                    </Select>
+                </FormGroup>
+                <FormGroup label={_("URL")}
+                    fieldId="edit-remote-url"
+                >
+                    <TextInput id="edit-remote-url"
+                        value={newURL}
+                        onChange={(_ev, url) => setNewURL(url)}
+                        type="text" />
+                </FormGroup>
+                <FormGroup fieldId="edit-remote-trusted">
+                    <Checkbox label={_("Use trusted GPG key")}
+                        id="gpg-verify"
+                        isChecked={isTrusted}
+                        onChange={(_ev, checked) => {
+                            setIsTrusted(checked);
+                        }} />
+                </FormGroup>
+                <FormGroup fieldId="add-another-key">
+                    {!addAnotherKey
+                        ? <Button isInline variant="secondary" id='add-another-key' onClick={() => setAddAnotherKey(true)}>{_("Add another key")}</Button>
+                        : <TextArea id='gpg-data'
+                                placeholder={ cockpit.format(_("Begins with $0"), "'-----BEGIN GPG PUBLIC KEY BLOCK-----'") }
+                                onChange={(_ev, key) => setKey(key)}
+                                value={key}
+                                aria-label={_("GPG public key")} />}
+                </FormGroup>
+            </Form>
+        </Modal>
+    );
+};
+
+EditRepositoryModal.propTypes = {
+    remote: PropTypes.string.isRequired,
+    availableRemotes: PropTypes.array.isRequired,
+};
+
+export const RebaseRepositoryModal = ({ origin, availableRemotes, currentOriginBranches, currentBranchLoadError, onChangeBranch, onChangeRemoteOrigin }) => {
+    const Dialogs = useDialogs();
+    const [isSelectRemoteExpanded, setSelectRemoteExpanded] = useState(false);
+    const [selectedRemote, setSelectedRemote] = useState(origin.remote);
+    const [isSelectBranchExpanded, setSelectBranchExpanded] = useState(false);
+    const [selectedBranch, setSelectedBranch] = useState(origin.branch);
+    const [availableBranches, setAvailableBranches] = useState(currentOriginBranches);
+    const [branchLoadError, setBranchLoadError] = useState(currentBranchLoadError);
+    const [loadingBranches, setLoadingBranches] = useState(false);
+    const [error, setError] = useState(null);
+
+    const handeRemoteSelect = async (remote) => {
+        setSelectedRemote(remote);
+        setLoadingBranches(true);
+        remotes.listBranches(remote)
+                .then(newBranches => {
+                    setBranchLoadError(null);
+                    setAvailableBranches(newBranches);
+                    setLoadingBranches(false);
+
+                    if (newBranches.includes(origin.branch)) {
+                        setSelectedBranch(origin.branch);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231124-085450-59247a42-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)